sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/src/Customized/Models/DataSourceInfo.cs

```diff
@@ -30,16 +30,19 @@ public DataSourceInfo(ResourceIdentifier resourceId)
         /// <param name="resourceLocation"> Location of datasource. </param>
         /// <param name="resourceName"> Unique identifier of the resource in the context of parent. </param>
         /// <param name="resourceType"> Resource Type of Datasource. </param>
-        /// <param name="resourceUri"> Uri of the resource. </param>
-        internal DataSourceInfo(string dataSourceType, string objectType, ResourceIdentifier resourceId, AzureLocation? resourceLocation, string resourceName, ResourceType? resourceType, Uri resourceUri)
+        /// <param name="resourceUriString"> Uri of the resource. </param>
+        internal DataSourceInfo(string dataSourceType, string objectType, ResourceIdentifier resourceId, AzureLocation? resourceLocation, string resourceName, ResourceType? resourceType, string resourceUriString)
         {
             DataSourceType = dataSourceType;
             ObjectType = objectType;
             ResourceId = resourceId;
             ResourceLocation = resourceLocation;
             ResourceName = resourceName;
             ResourceType = resourceType;
-            ResourceUri = resourceUri;
+            ResourceUriString = resourceUriString;
+#pragma warning disable CS0618 // Type or member is obsolete
+            ResourceUri = null;
+#pragma warning restore CS0618 // Type or member is obsolete
```

Comment:
We do not need to assign anything to this. When uninitialized, a property is `null`

Reply:
Fixed